platforms.go

```diff
@@ -174,9 +177,12 @@ func ParseAll(specifiers []string) ([]specs.Platform, error) {
 
 // Parse parses the platform specifier syntax into a platform declaration.
 //
-// Platform specifiers are in the format `<os>|<arch>|<os>/<arch>[/<variant>]`.
+// Platform specifiers are in the format `<os>[(<OSVersion>)]|<arch>|<os>[(<OSVersion>)]/<arch>[/<variant>]`.
 // The minimum required information for a platform specifier is the operating
-// system or architecture. If there is only a single string (no slashes), the
+// system or architecture. The OSVersion can be part of the OS like windows(10.0.17763)
+// Therefore, an os version is specified, then specs.Platform.OSVersion is populated.
+// If not it is left empty.
```

Comment:
More nits. Also, I wonder if necessary to explicitly mention that `os()` is equivalent to `os`? Or is that clear-enough from the description and test coverage?

```suggestion
// system or architecture. The OSVersion can be part of the OS like `windows(10.0.17763)`.
// When an OSVersion is specified, then specs.Platform.OSVersion is populated with that value,
// and an empty string otherwise.
```